src/core/positioning/positioning.cpp

```diff
@@ -534,6 +540,44 @@ void Positioning::processGnssPositionInformation()
 {
   mPositionInformation = mPositioningSourceReplica->property( "positionInformation" ).value<GnssPositionInformation>();
 
+  if ( mAveragedPosition )
+  {
+    mCollectedPositionInformations << mPositionInformation;
+    mPositionInformation = PositioningUtils::averagedPositionInformation( mCollectedPositionInformations );
+    emit averagedPositionCountChanged();
+  }
+
+  GnssPositionInformation::AccuracyQuality quality = GnssPositionInformation::AccuracyBad;
+  const double hacc = mPositionInformation.hacc();
+  const bool isExcellentThresholdDefined = !std::isnan( excellentAccuracyThreshold() );
+  const bool isBadThresholdDefined = !std::isnan( badAccuracyThreshold() );
+
+  if ( !std::isnan( hacc ) )
+  {
+    if ( isExcellentThresholdDefined && hacc <= excellentAccuracyThreshold() )
+    {
+      quality = GnssPositionInformation::AccuracyExcellent;
+    }
+    else if ( isBadThresholdDefined && hacc >= badAccuracyThreshold() )
+    {
+      quality = GnssPositionInformation::AccuracyOk;
+    }
+    else if ( isExcellentThresholdDefined || isBadThresholdDefined )
+    {
+      quality = GnssPositionInformation::AccuracyBad;
+    }
+    else
+    {
+      quality = GnssPositionInformation::AccuracyOk;
+    }
+  }
+  else
+  {
+    quality = GnssPositionInformation::AccuracyBad;
+  }
```

Comment:
We shouldn't classify as bad if hacc is nan _unless_ we are classifying. 

So here it should go:

```suggestion
  if ( isExcellentThresholdDefined && isBadThresholdDefined )
  {
    if ( !std::isnan )
    {
      if ( hacc <= excellentAccuracyThreshold() ) {
        quality = GnssPositionInformation::AccuracyExcellent;
      }
      else if ( isExcellentThresholdDefined || isBadThresholdDefined )
      {
        quality = GnssPositionInformation::AccuracyBad;
      }
      else
      {
        quality = GnssPositionInformation::AccuracyOk;
      }
    }
    else
    {
      quality = GnssPositionInformation::AccuracyBad;
     }  
  else
  {
    quality = GnssPositionInformation::AccuracyUndetermined;
  }
```

Reply:
we need to add this logic too right? 
```
// .. excellent
else if ( hacc <= badAccuracyThreshold() )
{
  quality = GnssPositionInformation::AccuracyOk;
}
```

Reply:
Yeah the logic above is wrong, I was typing in github web app :)